tests/end2end-tests/data_process/data_gen.py

```diff
@@ -56,6 +44,8 @@ def write_data_parquet(data, data_file):
 node_id3_str = np.array([str(nid) for nid in node_id3])
 node_data3 = {
```

Comment:
Can you add some docstring about how does the test graph looks like?

Reply:
i don't think there is any particular purpose of each node type. I create multiple node types to handle different cases, for example, to store node data with different formats, with or without node features, with or without labels, features of different types, node features with different dimensions. you can easily see this kind of information in the test code when node data are defined. I'm not sure what information should be documented.